sql/search-cases-query.sql

```diff
@@ -0,0 +1,53 @@
+SELECT DISTINCT ON (cases.id)
```

Comment:
I would agree with @GPaoloni's [comment](https://github.com/tech-matters/hrm/pull/54#pullrequestreview-5246420650) that we add some commenting and modify the indentation.  Though this is an epic and impressive query which taught me a few new Postgres tricks.  Nice.  Some integration tests may be a good idea too.